### PR TITLE
Support for PG15.

### PR DESCRIPTION
**Pull request by sb230132**
As part of this patch, added and fixed some of the regress checks which fail on PG15.

### CODE REVIEW

**Review by svenklemm — CHANGES_REQUESTED** [review]
The patch that adds test output for PG15 should also be the patch thats adds it to CI

**Comment by mfundul** [comment]
Not all tests pass yet, so no CI is possible.

File: .github/gh_matrix_builder.py [hunk 1]
```diff
@@ -120,6 +120,8 @@ def macos_config(overrides):
...
...agg_drop_chunks cagg_multi_dist_ht cagg_bgw_dist_ht cagg_ddl_dist_ht continuous_aggs continuous_aggs_deprecated  dist_util remote_txn telemetry_stats rowsecurity-15 pg_dump_unprivileged concurrent_add_dimension cagg_dump cagg_migrate_integer_dist_ht cagg_migrate_timestamp_dist_ht cagg_query-15 chunk_api compression compression_bgw compression_hypertable  cagg_compression compression_dml build_info memoize metadata parallel pg_dump plan_hashagg query dist_api_calls data_node debug_notice dist_commands dist_compression dist_copy_format_long dist_copy_long dist_ddl dist_move_chunk dist_partial_agg dist_policy dist_query exp_cagg_monthly exp_cagg_origin exp_cagg_timezone modify_exclusion partialize_finalize skip_scan transparent_decompression_queries constify_now-15 constify_timestamptz_op_interval constraint_exclusion_prepared decompress_placeholdervar dist_chunk dist_distinct dist_distinct_pushdown dist_fetcher_type dist_gapfill dist_insert dist_parallel_agg dist_queries dist_remote_error subtract_integer_from_now timestamp_limits ordered_append-15 ordered_append_join-15 space_constraint-15 transparent_decompress_chunk-15 jit-15 decompress_join-15 gapfill-15 generated_columns-15'"}))
```
Comment (svenklemm) [inline .github/gh_matrix_builder.py:123]: hmm why does this list ignore the output of the -15 tests you just added this means you added files that dont match the output on pg15

Comment (sb230132) [reply .github/gh_matrix_builder.py]: Reduced the number of -15 tests.
Only left are dist_hypertable-15, rowsecurity-15, jit-15

Created PR for different issues and grouped all tests with similar failures.

File: CMakeLists.txt [hunk 2]
```diff
@@ -323,9 +323,9 @@ set(PG_VERSION "${PG_VERSION_MAJOR}.${PG_VERSION_MINOR}")
 # Ensure that PostgreSQL version is supported and consistent with src/compat.h
 # version check
 if((${PG_VERSION_MAJOR} LESS "12")
-   OR (${PG_VERSION_MAJOR} GREATER "14")
+   OR (${PG_VERSION_MAJOR} GREATER "15")
    AND NOT (${EXPERIMENTAL}))
-  message(FATAL_ERROR "TimescaleDB only supports PostgreSQL 12, 13 and 14")
+  message(FATAL_ERROR "TimescaleDB only supports PostgreSQL 12, 13, 14 and 15")
```
Comment (svenklemm) [inline CMakeLists.txt:328]: I dont think this should be part of this patch instead we should make CI run with -DEXPERIMENTAL=ON. We should only enable this once PG15 fully works.

File: .github/gh_matrix_builder.py [hunk 3]
```diff
@@ -120,6 +120,20 @@ def macos_config(overrides):
 
 m["include"].append(build_without_telemetry({"pg":PG14_LATEST}))
 
+m["include"].append(build_debug_config({"pg":15,"snapshot":"snapshot", 
+# below tests are tracked as part of #4838
+      "installcheck_args": "SKIPS='003_connections_privs 001_simple_multinode 004_multinode_rdwr_1pc data_node_bootstrap dist_hypertable-15' "
```
Comment (svenklemm) [inline .github/gh_matrix_builder.py:125]:
```suggestion
m["include"].append(build_debug_config({"pg":15,"snapshot":"snapshot", "tsdb_build_args":"-DASSERTIONS=ON -DREQUIRE_ALL_TESTS=ON -DEXPERIMENTAL=ON"
# below tests are tracked as part of #4838
      "installcheck_args": "SKIPS='003_connections_privs 001_simple_multinode 004_multinode_rdwr_1pc data_node_bootstrap dist_hypertable-15' "
```

**Review by svenklemm — APPROVED** [review]

**Review by mfundul — APPROVED** [review]

**Comment by pantonis** [comment]
any update? When is support for pg15 is gonna be out?

**Pull request merged**